frontend/src/context/UserContext.tsx

```diff
@@ -0,0 +1,62 @@
+/* eslint-disable react-refresh/only-export-components */
+import {
+  createContext,
+  useContext,
+  useEffect,
+  useState,
+} from "react";
+import { InfoResponse, UserInfo } from "../types";
+
+interface UserContextType {
+  info: UserInfo | null;
+  loading: boolean;
+  logout: () => void;
+}
+
+const UserContext = createContext({} as UserContextType);
+
+export function UserProvider({ children }: { children: React.ReactNode }) {
+  const [info, setInfo] = useState<UserContextType["info"]>(null);
+  const [loading, setLoading] = useState(true);
+
+  const logout = async () => {
```

Comment:
Maybe I would just move this function out of the provider and do a hard navigation back to the home page on logout